### PR TITLE
Review feature

### PR DESCRIPTION
**Pull request by DanJFox86**
Implemented styled components.
https://trello.com/c/lXtLFT58

Added search component. Added route for searches.
https://trello.com/c/8r8WYNxk

Updated star rating for reviews.

### CODE REVIEW

File: client/src/components/Search.jsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                  value={this.props.sortOption}>
+          {sortOptions}
+          </select>
+        </SortOptions>
+      </SearchContainer>
+    );
+  }
+
+};
+export default Search;
```
Comment (presw) [inline client/src/components/Search.jsx]: Non-blocking: missing newline.

File: client/src/index.jsx [hunk 2]
```diff
@@ -32,6 +34,11 @@ class App extends React.Component {
     $.post('/api/reviews/vote', { info }, this.updateVotes.bind(this));
   }
 
+  searchQuery(e) {
+    console.log(`search parameters: `, e);
+
+  }
```
Comment (presw) [inline client/src/index.jsx]: Is this presently used for anything other than debugging?

File: Tests/test.js [hunk 3]
```diff
@@ -55,24 +55,25 @@ describe('Review: Testing DOM renders', () => {
...
   it('Review text should be rendered properly', () => {
     const wrapper = shallow(<Review review={review} voteClick={onVoteClick} />);
-    expect(wrapper.find('.review-text').text()).toEqual('Hi I am a review. I sorta liked this place, food was good. I suggest you go here.');
+    expect(wrapper.find('ReviewText').text()).toEqual('Hi I am a review. I sorta liked this place, food was good. I suggest you go here.');
   });
 
   // ------------------ USEFUL TEST -----------------------------
   it('Useful rating should be rendered properly', () => {
     const wrapper = shallow(<Review review={review} voteClick={onVoteClick} />);
+    console.log(JSON.stringify(wrapper.find('ReviewVote')));
```
Comment (aldajani) [inline Tests/test.js]: Extra console-log?

File: Tests/test.js [hunk 4]
```diff
@@ -89,41 +90,41 @@ describe('Review: Testing DOM renders', () => {
   });
 
   // ------------------ PHOTOS TEST -----------------------------
-  it('Cool rating should be rendered properly', () => {
-    const wrapper = shallow(<Review review={review} voteClick={onVoteClick} />);
-    expect(wrapper.find('.review-vote').find('.cool').text()).toEqual('Cool: 34');
-  });
+  // it('Cool rating should be rendered properly', () => {
```
Comment (aldajani) [inline Tests/test.js]: Is this meant to be commented out?

File: client/src/components/Review.jsx [hunk 5]
```diff
@@ -1,6 +1,134 @@
...
 
+const OrangeStar = styled.span`
+  color: orange;
+`;
+const Star = styled.span`
+  color: black;
+`;
+
+const ReviewContainer = styled.div`
+  border-top: solid 1px lightgray;
```
Comment (aldajani) [inline client/src/components/Review.jsx]: May want to consider separating out some of these components into their own files. It's getting a bit long, so just a suggestion. Something you could refactor later.

File: client/src/components/ReviewList.jsx [hunk 6]
```diff
@@ -1,5 +1,6 @@
 import React from 'react';
 import Review from './Review.jsx';
+import styled from 'styled-components'
```
Comment (aldajani) [inline client/src/components/ReviewList.jsx]: Do you need to import this?

File: client/src/components/Search.jsx [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+  }
+
+  onSearchChange(e) {
+    this.setState({
+      searchQuery: e.target.value,
+    });
+  }
+
+  executeSearch() {
+    this.props.search(this.state);
```
Comment (aldajani) [inline client/src/components/Search.jsx]: Did you mean to execute the search on the entire state of the component?

File: client/src/index.jsx [hunk 8]
```diff
@@ -1,8 +1,9 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
 import $ from 'jquery';
-// import Search from './components/Search.jsx';
+import Search from './components/Search.jsx';
 import ReviewList from './components/ReviewList.jsx';
+import styled from 'styled-components'
```
Comment (aldajani) [inline client/src/index.jsx]: Don't need to import if you aren't using it

File: client/src/components/Search.jsx (hunk 1 above)
Comment (DanJFox86) [reply client/src/components/Search.jsx]: corrected

File: client/src/index.jsx (hunk 2 above)
Comment (DanJFox86) [reply client/src/index.jsx]: Just placeholder, I think Josh told us to do everything up to writing the actual POST request.

File: Tests/test.js (hunk 4 above)
Comment (DanJFox86) [reply Tests/test.js]: Nah, deleted.

File: Tests/test.js (hunk 3 above)
Comment (DanJFox86) [reply Tests/test.js]: Yes, deleted.

File: client/src/components/Review.jsx (hunk 5 above)
Comment (DanJFox86) [reply client/src/components/Review.jsx]: Perhaps.

File: client/src/components/ReviewList.jsx (hunk 6 above)
Comment (DanJFox86) [reply client/src/components/ReviewList.jsx]: Not really, just dumped it in all the jsx files. Just removed it from the files that don't need it.

File: client/src/components/Search.jsx (hunk 7 above)
Comment (DanJFox86) [reply client/src/components/Search.jsx]: Yes. If I build out more search parameters and include them in the state, this function can remain the same.

File: client/src/index.jsx (hunk 8 above)
Comment (DanJFox86) [reply client/src/index.jsx]: Removed.

**Pull request merged**